src/save.ts

```diff
@@ -29,7 +29,35 @@ async function run(): Promise<void> {
             return;
         }
 
-        if (utils.isExactKeyMatch(primaryKey, state)) {
+        const envVarSet =
```

Comment:
As I understand it, this is the current behavior:

| `Inputs.UpdateEnvVariable` | `process.env[core.getInput(Inputs.UpdateEnvVariable)]` | Result |
| --- | --- | --- |
| Unset | Unset | Default Cache Behavior |
| Set | Unset or empty string | Default Cache Behavior |
| Set | `true` or `yes` | Update Cache |
| Set | Any non-empty string that is not `true` or `yes` | Skip Saving Cache |

In the second case, where the input is set but the env var hasn't been set, it falls back to the default cache behavior. Is that expected?  Just want to confirm...as I could also see that falling into the "Skip Saving Cache" result.

Reply:
The table that you have written is correct.

Reply:
environment variables don't allow for multiple, independently controlled choice of update.

e.g., two actions/cache -- one for tools, one with results of using those tools ... environment variable would affect both ... which is NOT a great solution.

Reply:
Why should that be true? You can use two different environment variables and they will function independently. Eg, `TOOLS_UPDATE` and `RESULTS_UPDATE`.